TestModels/Resource/runtimes/rust/src/types/i_simple_resource.rs

```diff
@@ -0,0 +1,56 @@
+// Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+
+pub trait ISimpleResource {
+    fn GetResourceData(
+        &mut self,
+        input: crate::operation::get_resource_data::GetResourceDataInput,
+    ) -> Result<
+        crate::operation::get_resource_data::GetResourceDataOutput,
+        crate::operation::get_resource_data::GetResourceDataError,
+    >;
+}
+
+#[derive(::std::clone::Clone, ::std::cmp::PartialEq, ::std::fmt::Debug)]
+pub struct ISimpleResourceObject {
```

Comment:
Can we reduce the visibility of the implementing struct somehow, since customer code should not have to reference it directly?

I'm also inclined to name it `SimpleResourceImpl` instead as "object" doesn't convey the right meaning for me.